pkg/container-disk/container-disk_test.go

```diff
@@ -272,36 +300,50 @@ var _ = Describe("ContainerDisk", func() {
 					SupportContainerResources: []v1.SupportContainerResources{},
 				})
 
-				vmi := api.NewMinimalVMI("fake-vmi")
-				appendContainerDisk(vmi, "r0")
+				By("Creating a new VMI with a container disk")
+				vmi := libvmi.New(
+					libvmi.WithContainerDisk("r0", "someimage:v1.2.3.4"),
+				)
+
+				By("Generating containers with the given VMI and cluster configuration")
 				containers := GenerateContainers(vmi, clusterConfig, nil, "libvirt-runtime", "/var/run/libvirt")
 
+				By("Setting the expected ephemeral storage request")
 				expectedEphemeralStorageRequest := resource.MustParse(ephemeralStorageOverheadSize)
 
 				containerResourceSpecs := make([]k8sv1.ResourceList, 0)
 				for _, container := range containers {
 					containerResourceSpecs = append(containerResourceSpecs, container.Resources.Requests)
 				}
 
+				By("Verifying that each container's resource spec includes the expected ephemeral storage request")
 				for _, containerResourceSpec := range containerResourceSpecs {
 					Expect(containerResourceSpec).To(HaveKeyWithValue(k8sv1.ResourceEphemeralStorage, expectedEphemeralStorageRequest))
 				}
 			})
+
 			It("by verifying container generation", func() {
 				clusterConfig, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{
 					SupportContainerResources: []v1.SupportContainerResources{},
 				})
-				vmi := api.NewMinimalVMI("fake-vmi")
-				appendContainerDisk(vmi, "r1")
-				appendContainerDisk(vmi, "r0")
+
+				By("Creating a new VMI with multiple container disks")
+				vmi := libvmi.New(
+					libvmi.WithContainerDisk("r1", "someimage:v1.2.3.4"),
+					libvmi.WithContainerDisk("r0", "someimage:v1.2.3.4"),
+				)
+
+				// Explicitly set ImagePullPolicy for each container disk
+				vmi.Spec.Volumes[0].ContainerDisk.ImagePullPolicy = k8sv1.PullAlways
```

Comment:
What is the value that `libvmi.WithContainerDisk` is using by default? I see `appendContainerDisk` used `PullAlways`.

Reply:
i needed to set ImagePullPolicy to something because without it i am getting the nil error
```go
vmi.Spec.Volumes[0].ContainerDisk.ImagePullPolicy = k8sv1.PullAlways
```

Reply:
I would add a new field in the method wit the pull policy, the goal here is to avoid to directly access the volumes

Reply:
so do i need to create new option ? 
like 
```go
libvmi.WithPullPolicy
```